lib/train/transports/ssh.rb

```diff
@@ -85,6 +85,15 @@ def connection(state = {}, &block)
 
     private
 
+    def reusable_connection?(conn_opts)
+      return false unless @connection_options
+
+      current_options = @connection_options
+      current_options.reject! { |k| %i{append_all_supported_algorithms}.include? k }
```

Comment:
`append_all_supported_algorithms` is needed in the ssh options to allow connections to use older and less secure algorithms. Is there a reason it gets discarded here?

Reply:
I will have a look at this one. But when I worked on this, it appeared that that option was removed somewhere on the current connection options. I'll come back to this

Reply:
(if it's only intended to be excluded from the comparison between the two option hashes, then line 91 should maybe use `current_options = @connection_options.dup` to take a copy of the options instead of a reference to `@connection_options`)

Reply:
Isn't the bug that the newly added `append_all_supported_algorithms` option is missing from the connection_options hash at https://github.com/tecracer-theinen/train/blob/master/lib/train/transports/ssh.rb#L145 ?